### ISSUE TITLE
HumanML3D reproduction: released checkpoint results are far from the reported paper numbers

### ISSUE THREAD

**Issue by dcx1110** [ChenFengYe/motion-latent-diffusion#67]
Hi, thank you for releasing the code and checkpoints.

We are currently trying to reproduce the HumanML3D text-to-motion results of MLD. However, we observed a large discrepancy between the results reported in the paper, the results obtained from the released official checkpoint, and the results from our own retrained model.

Our main purpose here is to verify whether we are following the intended training and evaluation protocol correctly, rather than to claim that there is an issue in the repository.

## Our setup

- Dataset: HumanML3D
- Task: text-to-motion
- Commit / branch: unavailable in our local copy, since the current workspace is not a git repository
- Environment:
  - Python: 3.10.20
  - PyTorch: 2.10.0+cu128
  - PyTorch Lightning: 1.9.5
  - Transformers: 5.3.0
  - Diffusers: 0.37.0
  - NumPy: 1.26.4
  - NVIDIA driver: 580.105.08
- GPU(s): 8 × NVIDIA GeForce RTX 5090 (32607 MiB each)

## Training command (for our retrained model)

```bash
python -m train \
  --cfg configs/config_mld_humanml3d.yaml \
  --cfg_assets configs/assets.yaml \
  --batch_size 64 \
  --nodebug \
  --device 0 1 2 3
```

Evaluation command
```python -m test \
  --cfg configs/config_mld_humanml3d_official_model20.yaml \
  --cfg_assets configs/assets.yaml
```

## Result comparison

For clarity, we summarize the three sources of results side by side below:

- **Paper (Table 9)**: reported numbers in the paper
- **Our retrained model**: trained from scratch using the public training pipeline
- **Released official checkpoint**: evaluated using the public evaluation pipeline

| Metric | Paper: MLD-1 (z0) | Paper: MLD-1 (ε) | Our retrained model | Released official checkpoint |
|---|---:|---:|---:|---:|
| R-Precision Top-1 | 0.447 ± 0.002 | 0.481 ± 0.003 | 0.29482758045196533 | 0.041821120493114 ± 0.0012493293299263995 |
| R-Precision Top-2 | 0.633 ± 0.002 | 0.673 ± 0.003 | 0.4710488518079122 | 0.0840086206793785 ± 0.0018172630701672357 |
| R-Precision Top-3 | 0.734 ± 0.002 | 0.772 ± 0.002 | 0.5877873500188192 | 0.12495689615607261 ± 0.0019480562148519055 |
| FID | 0.513 ± 0.011 | 0.473 ± 0.013 | 2.986289898554484 | 17.49972496032715 ± 0.1730752602671005 |
| MM Dist / Matching Score | 3.384 ± 0.008 | 3.196 ± 0.010 | 4.401715437571208 | 9.106504106521607 ± 0.017102073660891393 |
| Diversity | 9.181 ± 0.065 | 9.724 ± 0.082 | 8.694650967915853 | 8.107376790046692 ± 0.10763134981956735 |
| MModality / Multimodality | 0.735 ± 0.055 | 2.413 ± 0.079 | 3.173509438832601 | 6.030475997924805 ± 0.1870339521246243 |

## Additional reference metrics from the same official-checkpoint run

| Metric | Value |
|---|---:|
| gt R-Precision Top-1 | 0.5125538855791092 ± 0.0019492618387150365 |
| gt R-Precision Top-2 | 0.7022521585226059 ± 0.0015977874431254628 |
| gt R-Precision Top-3 | 0.7976616382598877 ± 0.0020938884206445395 |
| gt Diversity | 9.523457622528076 ± 0.09325567607031039 |

## What seems unusual to us

The discrepancy appears in **both** settings:

1. Our **retrained model** still does not match the paper results.
2. The **released official checkpoint** also performs far from the paper results in our environment.

So the issue does not seem to be only that our retraining failed to reproduce the paper. Even the released checkpoint gives results that are dramatically different from the reported numbers.

Because of this, we suspect there may be some **environment / dependency / evaluation compatibility issue**, or some missing training / evaluation details not fully reflected in the public repository.

Any clarification would be very helpful for reproduction. Thank you.
